Add build tags to many commands

Carpetsmoker opened the pull request:
Fixes #1464

fatih commented:
We should keep `:GoBuildTags`. :GoBuildTags is dynamic, and sometimes it's handy to have it. `go_build_tags` on the other hand is permanent, for people who always work on files that have it stick in their files.

Carpetsmoker commented:
> :GoBuildTags is dynamic, and sometimes it's handy to have it. go_build_tags on the other hand is permanent, for people who always work on files that have it stick in their files.

What do you mean with "dynamic" and "permanent" in this context? All `:GoBuildTags` does is set `go_build_tags`, so it behaves the same?

fatih commented:
Yeah I know that it sets it, but it's get resets when you restart vim. Hence the word dynamic. Of course it's not the best solution, because you override the prior value which was set via `go_build_tags`. The better solution would be to not override `go_build_tags` , but not sure if that is worth the complexity.  But the command is really handy, sometimes you see a new file that has a build tag and you want to change it quickly. I think we should have it, but maybe we should change how it behaves. Dunno.

bhcleek commented:
FWIW, I agree with @fatih; I use `:GoBuildTags` frequently. As a general guideline, anytime we expect users to be able to change a config value during a session, I'm in favor of providing a convenience function to set it.

Carpetsmoker commented:
Okay, fair enough. I added `:GoBuildTags` back. clarified the documentation, and rebased on mater after the config changes. Should be good to merge now.

Also fixes :GoBuildTags "" to clear the tags. Looks like this didn't work anymore (probably regression from Billy's config PR? Didn't investigate too deeply).

bhcleek commented on autoload/go/config.vim:
```diff
@@ -15,10 +15,8 @@ function! go#config#BuildTags() abort
-    if exists('g:go_build_tags')
-      unlet g:go_build_tags
-    endif
+  if a:value is '' || a:value is '""' || a:value is "''"
```
The conversion from `''` and `""` to the empty string should happen in `go#cmd#BuildTags`. I'll submit a separate PR to do that so it's resolved separately...

Carpetsmoker replied:
Alright, I merged your PR and rebased on master.

bhcleek commented:
assuming builds pass, lgtm

jdef commented:
My project doesn't use tags and I'm running golang 1.10. When I updated vim-go to `master` the `:GoCoverage` command complained that `tags` may only be specified once. When I rolled back `vim-go` to the commit just prior to his one, `:GoCoverage` worked again.
